src/Tasks/Microsoft.NET.Build.Tasks/AppHost.cs

```diff
@@ -99,18 +100,22 @@ private static int[] ComputeKMPFailureFunction(byte[] pattern)
         }
 
         // See: https://en.wikipedia.org/wiki/Knuth%E2%80%93Morris%E2%80%93Pratt_algorithm
-        private static int KMPSearch(byte[] pattern, byte[] bytes)
+        private static unsafe int KMPSearch(byte[] pattern, MemoryMappedViewAccessor accessor)
```

Comment:
Maybe use Span instead of going unsafe?

Reply:
reference to this comment  https://github.com/dotnet/sdk/pull/2293#discussion_r191953439 this code has to run on NET46

Reply:
We could pull in System.Memory, but then it's more dlls to distribute with the build task. I'd rather avoid the dependency as it seems each one we've pulled into net46 build task has caused one problem or another :(

Reply:
Yea, seems fine.